lib/compat/wordpress-6.3/html-api/class-gutenberg-html-tag-processor-6-3.php

```diff
@@ -1442,31 +1443,56 @@ private function apply_attributes_updates() {
 			$tail_delta = 0;
 
 			foreach ( $this->lexical_updates as $diff ) {
-				$update_head = $bookmark->start >= $diff->start;
-				$update_tail = $bookmark->end >= $diff->start;
+				$bookmark_start_is_after_diff_start = $bookmark->start >= $diff->start;
+				$bookmark_end_is_after_diff_end     = $bookmark->end >= $diff->start;
+
+				if ( $bookmark_start_is_after_diff_start ) {
+					$bookmark_end_is_before_diff_end = $bookmark->end < $diff->end;
+					if ( $bookmark_end_is_before_diff_end ) {
```

Comment:
I like using variables to assign names to computation, but this one seems a bit redundant. it's almost a literal reading of the computation, so maybe it would make just as much sense to leave this inline?

might also remove a layer of `if` nesting here since there's no use for it. there's _only_ an `A && B` condition here. even if we feel the need to leave a `movl eax, 14 # move 14 into eax`-style comment, we can extract it from the `if` without sacrificing performance.

```php
if ( $bookmark_start_is_after_diff_start && $bookmark->end < $diff->end ) {

}
```

```php
$bookmark_ends_inside_patch = $bookmark->end < $diff->end;

if ( $bookmark_starts_inside_patch && $bookmark_ends_inside_patch ) {

}
```

Reply:
played around some more with it and I'm less happy with the rename, particularly since they are _wrong_, including a mismatch between "is_after" and ">=", whereby it may not actually be _after_.

here's a tighter version stewing on relying on the existing names and using `continue 2` to move on to the next bookmark since we know at the point where we're removing the bookmark that we can skip any and all further processing for that bookmark.


not saying I love this version, but I find myself getting lost in the name and purpose of `$bookmark_start_is_after_diff_start`, finding myself wanting instead to move towards the semantic rather than the obvious, something like `$bookmark_shifts` or `$head_shifts` bringing me back to `$update_head`. so this does away with the names altogether; maybe we don't need them.

also I think if we bail before checking whether to release the bookmark we can skip once more some of the zig-zag processing of a bookmark (for release) that we don't need to (because it's unaffected). in other words, instead of repeating the question, "do we need to modify this bookmark?" we can bail immediately and then the rest of the body implies _yes we have to modify it else we would have bailed_.

interested in your thoughts

```php
			foreach ( $this->lexical_updates as $diff ) {
				if ( ! $bookmark->start >= $diff->start && ! $bookmark->end >= $diff->start ) {
					break;
				}

				if ( $bookmark->start >= $diff->start && $bookmark->end < $diff->end ) {
					$this->release_bookmark( $bookmark_name );
					continue 2;
				}

				$delta = strlen( $diff->text ) - ( $diff->end - $diff->start );

				if ( $bookmark->start >= $diff->start ) {
					$head_delta += $delta;
				}

				if ( $bookmark->end >= $diff->end ) {
					$tail_delta += $delta;
				}
			}
```

Reply:
I like it -- thank you for fiddling! 😊 

Sidenote: One thing that's a bit unfortunate with our merge-to-core process is that we don't have the full HTML Processor unit test suite in GB anymore. To verify that this doesn't break any existing tests, I guess I'll copy it to `wordpress-develop` and run unit tests there 😬

Reply:
> To verify that this doesn't break any existing tests, I guess I'll copy it to `wordpress-develop` and run unit tests there 😬 

Tests are passing 👍

Reply:
Applied the suggested change in f171f68adc.

Reply:
> we don't have the full HTML Processor unit test suite in GB anymore

yep, and this was a big concern of mine too 🤷‍♂️